wonk/models.py

```diff
@@ -153,10 +176,40 @@ def canonicalize_actions(actions: Set[str]) -> Union[str, List[str]]:
 def canonicalize_resources(resources: Set[str]) -> Union[str, List[str]]:
     """Return the set of resources as either a single string or a sorted list of strings."""
 
+    if "*" in resources:
+        return "*"
+
     if len(resources) == 1:
         return next(iter(resources))
 
-    return sorted(resources)
+    result = []
+    last_wildcard = None
```

Comment:
optional nit: Is there an easy way to indicate we stripped the asterisk? `last_wildcard_stem`?

Reply:
That's reasonable.